src/Network/Algorand/Node/Api.hs

```diff
@@ -62,6 +65,49 @@ data Version = Version
   deriving (Generic, Show)
 $(deriveJSON algorandSnakeOptions 'Version)
 
+newtype NanoSec = NanoSec { unNanoSec :: Integer }
+  deriving stock (Eq, Ord, Show)
+  deriving newtype (ToJSON, FromJSON)
```

Comment:
I would also probably derive `Integral` for it, just in case.

Reply:
Done.